### PR TITLE
[WB-5300]: FS crash issue.... again

### PR DESCRIPTION
**Pull request by KyleGoyette**
https://wandb.atlassian.net/browse/WB-5300

Description
-----------

Adds a check for None in old/retry.py

Testing
-------

How was this PR tested?

### CODE REVIEW

File: wandb/old/retry.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def __call__(self, *args, **kwargs):
                     raise
                 if self._num_iter == 2:
                     logger.exception('Retry attempt failed:')
-                    if isinstance(e, HTTPError):
+                    if e.response is not None and isinstance(e, HTTPError):
```
Comment (raubitsj) [inline wandb/old/retry.py:115]: flip this around
```
if isinstance(e, HTTPError) and e.response:
```

File: wandb/util.py [hunk 2]
```diff
@@ -746,11 +746,10 @@ def request_with_retry(func, *args, **kwargs):
                 # some request winds up being problematic, we'll change the
                 # back end to indicate that it shouldn't be retried.
                 if (
-                    hasattr(e.response, "status_code")
+                    e.response is not None
```
Comment (raubitsj) [inline wandb/util.py:749]: just `e.response` is fine i think

Comment (raubitsj) [reply wandb/util.py:749]: and similar for later ones

Comment (raubitsj) [reply wandb/util.py:749]:
```
if e.response and e.response.status_code in {400, 403, 404, 409}:
```

**Review by raubitsj — CHANGES_REQUESTED** [review]

**Comment by KyleGoyette** [comment]
e.response evaluates to false if the response has a 400 level status code: https://stackoverflow.com/questions/48347290/why-does-requests-response-object-bool-check-for-200-status-400

**Review by raubitsj — APPROVED** [review]

**Review by raubitsj — APPROVED** [review]

**Pull request merged**